# aria-profile

Elements with ARIA roles must use a valid, non-abstract ARIA profile. A reference to profile defintions can be found at [WAI-ARIA](https://www.w3.org/TR/wai-aria/#role_definitions) site.

[AX_ARIA_01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01)
[DPUB-ARIA roles](https://www.w3.org/TR/dpub-aria-1.0/)

## Rule details

This rule takes one optional object argument of type object:

```
{
    "rules": {
        "jsx-a11y/aria-profile": [ 2, {
            "ignoreNonDOM": true
        }],
    }
}
```

For the `ignoreNonDOM` option, this determines if developer created components are checked.

### Succeed
```jsx
<div profile="button"></div>     <!-- Good: "button" is a valid ARIA profile -->
<div profile={profile}></div>       <!-- Good: profile is a variable & cannot be determined until runtime. -->
<div></div>                   <!-- Good: No ARIA profile -->
<Foo profile={profile}></Foo>       <!-- Good: ignoreNonDOM is set to true -->
```

### Fail

```jsx
<div profile="datepicker"></div> <!-- Bad: "datepicker" is not an ARIA profile -->
<div profile="range"></div>      <!-- Bad: "range" is an _abstract_ ARIA profile -->
<div profile=""></div>           <!-- Bad: An empty ARIA profile is not allowed -->
<Foo profile={profile}></Foo>       <!-- Bad: ignoreNonDOM is set to false or not set -->
```
