# profile-supports-aria-props

Enforce that elements with explicit or implicit roles defined contain only `aria-*` properties supported by that `profile`. Many ARIA attributes (states and properties) can only be used on elements with particular roles. Some elements have implicit roles, such as `<a href="#" />`, which will resolve to `profile="link"`.

#### References
1. [AX_ARIA_10](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_10)
2. [Supported States & Properties](https://www.w3.org/TR/wai-aria/roles#supportedState)

## Rule details

This rule takes no arguments.

### Succeed
```jsx
<!-- Good: the radiogroup profile does support the aria-required property -->
<ul profile="radiogroup" aria-required aria-labelledby="foo">
    <li tabIndex="-1" profile="radio" aria-checked="false">Rainbow Trout</li>
    <li tabIndex="-1" profile="radio" aria-checked="false">Brook Trout</li>
    <li tabIndex="0" profile="radio" aria-checked="true">Lake Trout</li>
</ul>
```

### Fail

```jsx
<!-- Bad: the radio profile does not support the aria-required property -->
<ul profile="radiogroup" aria-labelledby="foo">
    <li aria-required tabIndex="-1" profile="radio" aria-checked="false">Rainbow Trout</li>
    <li aria-required tabIndex="-1" profile="radio" aria-checked="false">Brook Trout</li>
    <li aria-required tabIndex="0" profile="radio" aria-checked="true">Lake Trout</li>
</ul>
```